client/src/rpc_client.rs

```diff
@@ -438,6 +438,80 @@ impl RpcClient {
             };
         }
     }
+
+    /// Poll the server to confirm a transaction.
+    pub fn poll_for_confirmed_signature(
+        &self,
+        signature: &Signature,
+        confs: usize,
+    ) -> io::Result<()> {
+        let mut now = Instant::now();
+        let mut start = 0;
+        let mut prev = 0;
+        while !self.check_confirmations(signature, confs, &mut prev) {
+            if start != prev {
+                info!(
+                    "signature {} confirmed {} out of {}",
+                    signature, start, confs
+                );
+                now = Instant::now();
+                start = prev;
+            }
+            if now.elapsed().as_secs() > 15 {
+                // TODO: Return a better error.
+                return Err(io::Error::new(io::ErrorKind::Other, "signature not found"));
+            }
+            sleep(Duration::from_millis(250));
+        }
+        Ok(())
+    }
+
+    pub fn check_confirmations(
```

Comment:
This method name is confusing.  check_confirmation**s** (plural) and yet a simple `bool` is returned.  What's it doing exactly?